src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/runInTerminalTool.ts

```diff
@@ -220,9 +223,18 @@ export class RunInTerminalTool extends Disposable implements IToolImpl {
 			throw new Error('A chat session ID is required for this tool');
 		}
 
-		const command = toolSpecificData?.command ?? this._rewrittenCommand ?? args.command;
-		const didUserEditCommand = typeof toolSpecificData?.command === 'string' && toolSpecificData.command !== args.command;
-		const didToolEditCommand = !didUserEditCommand && this._rewrittenCommand !== undefined;
+		let command: string | undefined;
+		let didUserEditCommand: boolean;
+		let didToolEditCommand: boolean;
+		if (toolSpecificData.kind === 'terminal') {
+			command = toolSpecificData.command ?? this._rewrittenCommand ?? args.command;
+			didUserEditCommand = typeof toolSpecificData?.command === 'string' && toolSpecificData.command !== args.command;
+			didToolEditCommand = !didUserEditCommand && this._rewrittenCommand !== undefined;
+		} else {
+			command = toolSpecificData.commandLine.userEdited ?? toolSpecificData.commandLine.toolEdited ?? toolSpecificData.commandLine.original;
+			didUserEditCommand = toolSpecificData.commandLine.userEdited !== toolSpecificData.commandLine.original;
+			didToolEditCommand = !didUserEditCommand && toolSpecificData.commandLine.toolEdited !== toolSpecificData.commandLine.original;
```

Comment:
I might simplify this later, but I think having everything available will be useful in the future.